accelerator/cuda_accelerator.py

```diff
@@ -15,12 +15,28 @@
 except ImportError:
     pass
 
+# Delay import pynvml to avoid import error when CUDA is not available
+pynvml = None
+
 
 class CUDA_Accelerator(DeepSpeedAccelerator):
 
     def __init__(self):
         self._name = 'cuda'
         self._communication_backend_name = 'nccl'
+        if pynvml is None:
+            self._init_pynvml()
+
+    def _init_pynvml(self):
+        global pynvml
+        try:
+            import pynvml
+        except ImportError:
+            return
+        try:
+            pynvml.nvmlInit()
+        except pynvml.NVMLError:
+            return
```

Comment:
```suggestion
        global pynvml
        try:
            import pynvml as tmp_pynvml
        except ImportError:
            return
        try:
            tmp_pynvml.nvmlInit()
            pynvml = tmp_pynvml
        except tmp_pynvml.NVMLError:
            return
```

We would need to do something like this to avoid a bug when we are able to import `pynvml` by `pynvml.nvmlInit()` fails. For example, consider this:
```python
torch = None

def init_torch():
    global torch
    try:
        import torch
    except:
        return
    try:
        torch.fake_function()
    except:
        print("TORCH FAILED")
        return
        
init_torch()
if torch:
    print("TORCH IMPORTED")
```

Running this example outputs the following:
```bash
❯ python test.py  
TORCH FAILED
TORCH IMPORTED
```

This means that below in `available_memory`, we may fail to use the fallback method in the case where pynvml fails to initialize due to `if pynvml` on line 156.